### PR TITLE
Add bindings and helpers for C atomics

### PR DESCRIPTION
**Pull request by WojciechMazur**
* Add `C` atomic.h bindings for all primitive types and fences. 
* Restore usage of C atomics in `java.lang.Thread` and `java.lang.impl.PosixThread` implementation
* Add subset of c atomics in `nativelib`s `scala.scalanative.runtime.libc` to implement required fences in `NativeThread`

### CODE REVIEW

File: clib/src/main/resources/scala-native/atomic.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include <atomic>
+
+// Due to some bug (in GCC?) it might not be possible to access
```
Comment (ekrich) [inline clib/src/main/resources/scala-native/atomic.cpp:3]: I may be missing something but is this GCC or glibc?

File: javalib/src/main/scala/java/lang/impl/PosixThread.scala [hunk 2]
```diff
@@ -116,11 +118,7 @@ private[java] class PosixThread(val thread: Thread, stackSize: Long)
       isAbsolute: Boolean
   ): Unit = if (isMultithreadingEnabled) {
     // fast-path check, return if can skip parking
-    // TODO: atomics
-    val prev = counter
-    counter = 0
-    if (prev > 0) return
-    //  if (counterAtomic.exchange(0) > 0) return
+    if (counterAtromic.exchange(0) > 0) return
```
Comment (bishabosha) [inline javalib/src/main/scala/java/lang/impl/PosixThread.scala:121]:
```suggestion
    if (counterAtomic.exchange(0) > 0) return
```

File: javalib/src/main/scala/java/lang/impl/PosixThread.scala [hunk 3]
```diff
@@ -263,10 +261,9 @@ private[java] class PosixThread(val thread: Thread, stackSize: Long)
         fromRawPtr(elemRawPtr(base, offset))
     }
 
-// TODO: atomics
-//  @alwaysinline private def counterAtomic = new CAtomicInt(
-//    fromRawPtr(classFieldRawPtr(this, "counter"))
-//  )
+  @alwaysinline private def counterAtromic = new CAtomicInt(
```
Comment (bishabosha) [inline javalib/src/main/scala/java/lang/impl/PosixThread.scala:264]: not sure how many references there are to this definition
```suggestion
  @alwaysinline private def counterAtomic = new CAtomicInt(
```

File: javalib/src/main/scala/java/lang/impl/PosixThread.scala (hunk 2 above)
Comment (WojciechMazur) [reply javalib/src/main/scala/java/lang/impl/PosixThread.scala]: Fixed, thanks for catching the typo

File: clib/src/main/resources/scala-native/atomic.cpp (hunk 1 above)
Comment (WojciechMazur) [reply clib/src/main/resources/scala-native/atomic.cpp]: To be honest I don't remember what was the original cause, I was having these problems ~ 1-1.5 years ago. I've tried again to move these 2 definitions to `.c` file and it worked, so I've removed `atomic.cpp` entirely.

**Pull request merged**